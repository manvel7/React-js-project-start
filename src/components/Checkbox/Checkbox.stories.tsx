import React from "react"
import {Meta, Story} from "@storybook/react/types-6-0"
import Checkbox, {IProps} from "./Checkbox"

const meta: Meta<IProps> = {
  title: "Components/Checkbox",
  component: Checkbox,
  argTypes: {
    onChange: {action: "onChange"}
  }
}

const CheckboxStory: Story<IProps> = (args) => <Checkbox {...args} />

const DefaultCheckbox = CheckboxStory.bind({})
DefaultCheckbox.args = {
  label: "Default checkbox"
}

const DefaultChecked = CheckboxStory.bind({})
DefaultChecked.args = {
  label: "Default checked",
  checked: true
}

const DashedChecked = CheckboxStory.bind({})
DashedChecked.args = {
  label: "Dashed checkbox",
  checked: true,
  variant: "dashed"
}

const DefaultDisabled = CheckboxStory.bind({})
DefaultDisabled.args = {
  label: "Disabled checkbox",
  disabled: true,
  checked: true
}

const DashedDisabled = CheckboxStory.bind({})
DashedDisabled.args = {
  label: "Disabled dashed checkbox",
  checked: true,
  variant: "dashed",
  disabled: true
}

export {
  DefaultCheckbox,
  DefaultChecked,
  DashedChecked,
  DefaultDisabled,
  DashedDisabled
}

export default meta
