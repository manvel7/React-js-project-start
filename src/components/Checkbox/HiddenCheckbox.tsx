import {HTMLAttributes} from "react"
import styled from "styled-components"

export interface IProps extends HTMLAttributes<HTMLInputElement> {}

const HiddenCheckbox = styled.input.attrs({type: "checkbox"})<IProps>`
  border: 0;
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
`

export default HiddenCheckbox
