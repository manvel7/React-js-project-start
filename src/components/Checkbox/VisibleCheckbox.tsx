import {HTMLAttributes} from "react"
import styled, {css, StyledProps} from "styled-components"

export interface IProps extends HTMLAttributes<HTMLDivElement> {
  disabled?: boolean
  checked?: boolean
}

const disabledMixin = css<StyledProps<IProps>>`
  background: ${({theme}) => theme.palette.text.darkDimmed};
  border-color: ${({theme}) => theme.palette.disabled};
  color: ${({theme}) => theme.palette.disabled};
`

const VisibleCheckbox = styled.div<IProps>`
  align-items: center;
  background: ${({theme}) => theme.palette.background.white};
  border: 0.1rem solid ${({theme}) => theme.palette.primary};
  border-radius: 0.4rem;
  color: ${({theme}) => theme.palette.primary};
  display: flex;
  height: 1.6rem;
  justify-content: center;
  margin-right: 1.6rem;
  transition: all 150ms;
  width: 1.6rem;

  ${({disabled}) => (disabled ? disabledMixin : "")};

  svg {
    visibility: ${({checked}) => (checked ? "visible" : "hidden")};
  }
`

export default VisibleCheckbox
