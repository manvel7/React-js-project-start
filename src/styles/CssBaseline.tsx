import {
  createGlobalStyle,
  css,
  DefaultTheme,
  GlobalStyleComponent,
  StyledProps
} from "styled-components"
import {mediaBreakpointDown} from "../utils/breakpoint"
import {getValue, toPercent} from "../utils/unit"
import {
  bodyMixin,
  smallMixin,
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin
} from "./typography"

interface IProps {}

const genericResetMixin = css<StyledProps<IProps>>`
  html,
  body,
  div,
  span,
  applet,
  object,
  iframe,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  p,
  blockquote,
  pre,
  a,
  abbr,
  acronym,
  address,
  big,
  cite,
  code,
  del,
  dfn,
  em,
  img,
  ins,
  kbd,
  q,
  s,
  samp,
  small,
  strike,
  strong,
  sub,
  sup,
  tt,
  var,
  b,
  u,
  i,
  center,
  dl,
  dt,
  dd,
  ol,
  ul,
  li,
  fieldset,
  form,
  label,
  legend,
  table,
  caption,
  tbody,
  tfoot,
  thead,
  tr,
  th,
  td,
  article,
  aside,
  canvas,
  details,
  embed,
  figure,
  figcaption,
  footer,
  header,
  hgroup,
  menu,
  nav,
  output,
  ruby,
  section,
  summary,
  time,
  mark,
  audio,
  video {
    margin: 0;
    padding: 0;
    border: 0;
    vertical-align: baseline;
  }

  article,
  aside,
  details,
  figcaption,
  figure,
  footer,
  header,
  hgroup,
  menu,
  nav,
  section {
    display: block;
  }

  ol,
  ul {
    list-style: none;
  }

  blockquote,
  q {
    quotes: none;
  }

  blockquote:before,
  blockquote:after,
  q:before,
  q:after {
    content: "";
    content: none;
  }

  table {
    border-collapse: collapse;
    border-spacing: 0;
  }
`

const appResetMixin = css<StyledProps<IProps>>`
  html {
    /* stylelint-disable property-no-vendor-prefix */
    -ms-overflow-style: scrollbar; /* Set box-sizing so that the width is not affected by padding or border */
    -ms-text-size-adjust: 100%; /* Set default font family in all browsers. */
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); /* Makes 1rem = 10px */
    -webkit-text-size-adjust: 100%; /* Prevent adjustments of font size after orientation changes in IE on Windows Phone and in iOS. */
    /* stylelint-enable property-no-vendor-prefix */
    box-sizing: border-box;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }

  body {
    background-color: ${({theme}) => theme.palette.background.white};

    /* Hide overflow content */
    &.overflow {
      overflow: hidden;
    }

    /* Prevent focus outline on elements that can't be accessed via keyboard. */
    [tabindex="-1"]:focus {
      outline: none !important;
    }

    /* Hide overflow in IE */
    svg:not(:root) {
      overflow: hidden;
    }

    /* Hide elements has 'hidden' attribute in IE 10- */
    [hidden] {
      display: none !important;
    }

    ::-webkit-file-upload-button {
      /* stylelint-disable-next-line property-no-vendor-prefix */
      -webkit-appearance: button;
      font: inherit;
    }
  }
`

const typographyMixin = css<StyledProps<IProps>>`
  html {
    font-family: ${({theme}) =>
      theme.typography
        .fontFamily}; /* Prevent scrollbars to overlap content in IE11 and Edge */
    font-size: ${({theme}) =>
      toPercent(
        getValue(theme.typography.htmlFontSize),
        16
      )}; /* Set default tap highlight to be transparent in iOS. */
    font-style: normal;
    color: ${({theme}) => theme.palette.text.dark};
  }

  body {
    ${bodyMixin<IProps>()}
  }

  h1 {
    ${h1Mixin<IProps>()}
  }

  h2 {
    ${h2Mixin<IProps>()}
  }

  h3 {
    ${h3Mixin<IProps>()}
  }

  h4 {
    ${h4Mixin<IProps>()}
  }

  h5 {
    ${h5Mixin<IProps>()}
  }

  h6 {
    ${h6Mixin<IProps>()}
  }

  small {
    ${smallMixin<IProps>()}
  }

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.bold};

    h1,
    h2,
    h3,
    h4,
    h5,
    h6,
    caption {
      font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
    }
  }
`

const utilsMixin = css<StyledProps<IProps>>`
  .hidden {
    visibility: hidden;
  }

  .d-none {
    display: none;
  }

  .hidden-sm-down {
    ${({theme}) =>
      mediaBreakpointDown(
        "sm",
        theme.grid.breakpoints,
        `
      display: none;
    `
      )}
  }

  .text-center {
    text-align: center;
  }
`

const CssBaseline: GlobalStyleComponent<
  IProps,
  DefaultTheme
> = createGlobalStyle`
  ${genericResetMixin}

  ${appResetMixin}

  ${typographyMixin}

  ${utilsMixin}
`

export default CssBaseline
