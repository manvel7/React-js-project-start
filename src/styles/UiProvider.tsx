import React, {FunctionComponent} from "react"
import {ThemeProvider} from "styled-components"
import theme from "./theme"
import {CssBaseline} from "./index"

interface IProps {}

const UiProvider: FunctionComponent<IProps> = (props) => {
  const {children} = props

  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </ThemeProvider>
  )
}

export default UiProvider
