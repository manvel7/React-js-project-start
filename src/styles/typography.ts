import {
  css,
  DefaultTheme,
  FlattenInterpolation,
  StyledProps,
  ThemedStyledProps
} from "styled-components"
import {pxToRem} from "../utils/unit"

const h1Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h1.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h1.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h1.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h1.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h2Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h2.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h2.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h2.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h2.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h3Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h3.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h3.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h3.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h3.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h4Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h4.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h4.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h4.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h4.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h5Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h5.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h5.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h5.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h5.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const h6Mixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(theme.typography.variants.h6.size, theme.typography.htmlFontSize)};
  font-weight: ${({theme}) => theme.typography.variants.h6.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.h6.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.h6.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`

const bodyMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.body.weight};
  letter-spacing: ${({theme}) => theme.typography.variants.body.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.body.lineHeight,
      theme.typography.htmlFontSize
    )};
`

const smallMixin = <P extends Record<any, any>>(): FlattenInterpolation<
  ThemedStyledProps<P, DefaultTheme>
> => css<StyledProps<P>>`
  font-size: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.size,
      theme.typography.htmlFontSize
    )};
  font-weight: ${({theme}) => theme.typography.variants.caption.weight};
  letter-spacing: ${({theme}) =>
    theme.typography.variants.caption.letterSpacing};
  line-height: ${({theme}) =>
    pxToRem(
      theme.typography.variants.caption.lineHeight,
      theme.typography.htmlFontSize
    )};

  b,
  strong {
    font-weight: ${({theme}) => theme.typography.fontWeights.boldest};
  }
`
export {
  h1Mixin,
  h2Mixin,
  h3Mixin,
  h4Mixin,
  h5Mixin,
  h6Mixin,
  bodyMixin,
  smallMixin
}
