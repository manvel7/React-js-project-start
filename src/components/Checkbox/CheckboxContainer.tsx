import {HTMLAttributes} from "react"
import styled from "styled-components"
import {h6Mixin} from "../../styles/typography"

export interface IProps extends HTMLAttributes<HTMLElement> {
  disabled?: boolean
}

const CheckboxContainer = styled.label<IProps>`
  align-items: center;
  color: ${({disabled, theme}) =>
    disabled ? theme.palette.text.darkDimmed : theme.palette.text.dark};
  display: inline-flex;
  justify-content: center;
  ${h6Mixin<IProps>()};
`

export default CheckboxContainer
