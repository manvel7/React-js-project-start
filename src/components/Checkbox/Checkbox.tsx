import React, {FunctionComponent, HTMLAttributes, ReactNode} from "react"
import {faCheckSquare, faMinusSquare} from "@fortawesome/pro-solid-svg-icons"
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome"
import CheckboxContainer from "./CheckboxContainer"
import HiddenCheckbox from "./HiddenCheckbox"
import VisibleCheckbox from "./VisibleCheckbox"

export type CheckboxVariant = "default" | "dashed"

export interface IProps extends HTMLAttributes<HTMLInputElement> {
  checked?: boolean
  disabled?: boolean
  label?: ReactNode
  variant?: CheckboxVariant
}

const Checkbox: FunctionComponent<IProps> = (props) => {
  const {checked, label, disabled, variant, className, ...rest} = props

  return (
    <CheckboxContainer disabled={disabled} className={className}>
      <HiddenCheckbox checked={checked} disabled={disabled} {...rest} />
      <VisibleCheckbox checked={checked} disabled={disabled}>
        <FontAwesomeIcon
          icon={variant === "dashed" ? faMinusSquare : faCheckSquare}
        />
      </VisibleCheckbox>
      {label}
    </CheckboxContainer>
  )
}

Checkbox.defaultProps = {
  onChange: () => undefined
}

export default Checkbox
