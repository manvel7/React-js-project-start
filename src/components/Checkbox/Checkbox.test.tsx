import React from "react"
import {render, screen} from "@testing-library/react"
import {UiProvider} from "../../styles"
import theme from "../../styles/theme"
import Checkbox from "./Checkbox"

describe("Checkbox", () => {
  describe("Default", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Checkbox />
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Checkbox />
        </UiProvider>
      )

      const element = screen.getByRole("checkbox")
      expect(element).toBeInTheDocument()

      const component = result.container.firstChild
        ? result.container.firstChild.childNodes[1]
        : null
      expect(component).toHaveStyleRule("border-radius", "0.4rem")
    })
  })

  describe("Default checked", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Checkbox checked={true} />
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Checkbox checked={true} />
        </UiProvider>
      )

      const element = screen.getByRole("checkbox")
      expect(element).toBeInTheDocument()

      const iconElement = screen.getAllByRole("img", {hidden: true})[0]
      expect(iconElement).toBeInTheDocument()

      const component = result.container.firstChild
        ? result.container.firstChild.childNodes[1]
        : null
      expect(component).toHaveStyleRule("color", theme.palette.primary)
    })
  })

  describe("Disabled checked", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Checkbox checked={true} disabled={true} />
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Checkbox checked={true} disabled={true} />
        </UiProvider>
      )

      const element = screen.getByRole("checkbox")
      expect(element).toBeInTheDocument()

      const iconElement = screen.getAllByRole("img", {hidden: true})[0]
      expect(iconElement).toBeInTheDocument()

      const component = result.container.firstChild
        ? result.container.firstChild.childNodes[1]
        : null
      expect(component).toHaveStyleRule("color", theme.palette.disabled)
    })
  })

  describe("Dashed checkbox checked", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Checkbox checked={true} variant="dashed" />
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Checkbox checked={true} variant="dashed" />
        </UiProvider>
      )

      const element = screen.getByRole("checkbox")
      expect(element).toBeInTheDocument()

      const iconElement = screen.getAllByRole("img", {hidden: true})[0]
      expect(iconElement).toBeInTheDocument()

      const component = result.container.firstChild
        ? result.container.firstChild.childNodes[1]
        : null
      expect(component).toHaveStyleRule("color", theme.palette.primary)
    })
  })

  describe("Dashed disabled checkbox checked", () => {
    it("should match snapshot", () => {
      const result = render(
        <UiProvider>
          <Checkbox checked={true} disabled={true} variant="dashed" />
        </UiProvider>
      )
      expect(result.container.firstChild).toMatchSnapshot()
    })

    it("renders on screen properly", () => {
      const result = render(
        <UiProvider>
          <Checkbox disabled={true} checked={true} variant="dashed" />
        </UiProvider>
      )

      const element = screen.getByRole("checkbox")
      expect(element).toBeInTheDocument()

      const iconElement = screen.getAllByRole("img", {hidden: true})[0]
      expect(iconElement).toBeInTheDocument()

      const component = result.container.firstChild
        ? result.container.firstChild.childNodes[1]
        : null
      expect(component).toHaveStyleRule("color", theme.palette.disabled)
    })
  })
})
