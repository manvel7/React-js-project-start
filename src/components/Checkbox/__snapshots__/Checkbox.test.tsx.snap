// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Dashed checkbox checked should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #232323;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c1 {
  border: 0;
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FEFEFE;
  border: 0.1rem solid #001254;
  border-radius: 0.4rem;
  color: #001254;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.6rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 1.6rem;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  width: 1.6rem;
}

.c2 svg {
  visibility: visible;
}

<label
  class="c0"
>
  <input
    checked=""
    class="c1"
    type="checkbox"
  />
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-minus-square fa-w-14 "
      data-icon="minus-square"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM92 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H92z"
        fill="currentColor"
      />
    </svg>
  </div>
</label>
`;

exports[`Checkbox Dashed disabled checkbox checked should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #808080;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c1 {
  border: 0;
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FEFEFE;
  border: 0.1rem solid #001254;
  border-radius: 0.4rem;
  color: #001254;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.6rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 1.6rem;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  width: 1.6rem;
  background: #808080;
  border-color: #DDDDDD;
  color: #DDDDDD;
}

.c2 svg {
  visibility: visible;
}

<label
  class="c0"
  disabled=""
>
  <input
    checked=""
    class="c1"
    disabled=""
    type="checkbox"
  />
  <div
    class="c2"
    disabled=""
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-minus-square fa-w-14 "
      data-icon="minus-square"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM92 296c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h264c6.6 0 12 5.4 12 12v56c0 6.6-5.4 12-12 12H92z"
        fill="currentColor"
      />
    </svg>
  </div>
</label>
`;

exports[`Checkbox Default checked should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #232323;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c1 {
  border: 0;
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FEFEFE;
  border: 0.1rem solid #001254;
  border-radius: 0.4rem;
  color: #001254;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.6rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 1.6rem;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  width: 1.6rem;
}

.c2 svg {
  visibility: visible;
}

<label
  class="c0"
>
  <input
    checked=""
    class="c1"
    type="checkbox"
  />
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-check-square fa-w-14 "
      data-icon="check-square"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
        fill="currentColor"
      />
    </svg>
  </div>
</label>
`;

exports[`Checkbox Default should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #232323;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c1 {
  border: 0;
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FEFEFE;
  border: 0.1rem solid #001254;
  border-radius: 0.4rem;
  color: #001254;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.6rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 1.6rem;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  width: 1.6rem;
}

.c2 svg {
  visibility: hidden;
}

<label
  class="c0"
>
  <input
    class="c1"
    type="checkbox"
  />
  <div
    class="c2"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-check-square fa-w-14 "
      data-icon="check-square"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
        fill="currentColor"
      />
    </svg>
  </div>
</label>
`;

exports[`Checkbox Disabled checked should match snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #808080;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.6rem;
  font-weight: 600;
  -webkit-letter-spacing: 0em;
  -moz-letter-spacing: 0em;
  -ms-letter-spacing: 0em;
  letter-spacing: 0em;
  line-height: 1.6rem;
}

.c0 b,
.c0 strong {
  font-weight: 800;
}

.c1 {
  border: 0;
  height: 0.1rem;
  margin: -0.1rem;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 0.1rem;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FEFEFE;
  border: 0.1rem solid #001254;
  border-radius: 0.4rem;
  color: #001254;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.6rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 1.6rem;
  -webkit-transition: all 150ms;
  transition: all 150ms;
  width: 1.6rem;
  background: #808080;
  border-color: #DDDDDD;
  color: #DDDDDD;
}

.c2 svg {
  visibility: visible;
}

<label
  class="c0"
  disabled=""
>
  <input
    checked=""
    class="c1"
    disabled=""
    type="checkbox"
  />
  <div
    class="c2"
    disabled=""
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-check-square fa-w-14 "
      data-icon="check-square"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M400 480H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48zm-204.686-98.059l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.248-16.379-6.249-22.628 0L184 302.745l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.25 16.379 6.25 22.628.001z"
        fill="currentColor"
      />
    </svg>
  </div>
</label>
`;
